test/gtest/gtest_ck_common.hpp

```diff
@@ -0,0 +1,48 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2024 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+#include <gtest/gtest_common.hpp>
+
+#if MIOPEN_USE_COMPOSABLEKERNEL
+#define WORAROUND_ISSUE_2533 1
+#endif
+
+// MI100 : gfx908
+// MI200 : gfx90a
+// MI300 : gfx940, gfx941, gfx942
+static inline bool IsDeviceSupportedForCK()
+{
+#if MIOPEN_USE_COMPOSABLEKERNEL
+#if WORAROUND_ISSUE_2533
+    using e_mask = enabled<Gpu::gfx908, Gpu::gfx90A, Gpu::gfx94X>;
+    using d_mask = disabled<Gpu::gfx103X, Gpu::gfx110X, Gpu::gfx120X>;
+    return ::IsTestSupportedForDevMask<d_mask, e_mask>();
```

Comment:
```suggestion
    return IsTestSupportedByDevice(Gpu::gfx908 | Gpu::gfx90A | Gpu::gfx94X);
```

Reply:
Let me clarify this situation a bit.
`e_mask\d_mask` were introduced specifically to make CTest porting easy - old CTest infrastructure has those CMake flags and I wanted to implement exactly the same functionality for code equality verification - you can compare 1:1 CTest and GTest device lists. Under the hood it has some defaults, disable mask excludes some gpus from default mask and enable mask includes new gpus. The key functionality is a default mask under the hood.

For the new tests I guess it's better to feed a list of supported devices, instead of enabled\disabled masks.